mesonbuild/compilers/cpp.py

```diff
@@ -203,17 +203,19 @@ def language_stdlib_only_link_flags(self, env: Environment) -> T.List[str]:
         machine = env.machines[self.for_machine]
         assert machine is not None, 'for mypy'
 
-        # We need to determine whether to use libc++ or libstdc++. We can't
-        # really know the answer in most cases, only the most likely answer,
-        # because a user can install things themselves or build custom images.
-        search_order: T.List[str] = []
-        if machine.system in {'android', 'darwin', 'dragonfly', 'freebsd', 'netbsd', 'openbsd'}:
-            search_order = ['c++', 'stdc++']
-        else:
-            search_order = ['stdc++', 'c++']
-        for lib in search_order:
-            if self.find_library(lib, env, []) is not None:
-                return search_dirs + [f'-l{lib}']
+        # https://stackoverflow.com/a/31658120
+        try:
+            # If we can, use C++20's <version>.
+            self._find_best_cpp_std('c++20')
+            header = 'version'
+        except MesonException:
+            header = 'ciso646'
```

Comment:
Does it even matter if C++ has c++20 here, or can we just do something like (not sure what the method actually is):
```python
header ='version' if self.has_header('<version>') else 'cisco646'
```

Reply:
Nah, we can fall back fine if we do that in theory. Let me try it.. Thanks!

Reply:
Nice idea. That also means we can do a preprocess check for the header presence instead of a compile check to see if the -std option is a valid flag. In theory this is a bit faster.